src/frontend-samples/read-settings-sample/ExternalSettingsUI.tsx

```diff
@@ -0,0 +1,169 @@
+    }
+
+    const imodelQuery = new IModelQuery();
+    imodelQuery.byName(iModelName);
+    const imodels = await IModelApp.iModelClient.iModels.get(context, project.wsgId, imodelQuery);
+    if (imodels.length === 0)
+      throw new Error(`iModel with name "${iModelName}" does not exist in project "${projectName}"`);
+
+    return { projectId: project.wsgId, imodelId: imodels[0].wsgId };
+  }
+
+  // Handler to read external settings when name in dropdown changes
+  handleSettingsChange(event: ChangeEvent<HTMLSelectElement>) {
+    const settingKey = event.target.value;
+    this.setState({ settingKey: settingKey });
+
+    AuthorizedFrontendRequestContext.create().then(async context => {
+      const response = await IModelApp.settings.getSetting(context, namespace, settingKey, true, this.state.projectId, this.state.imodelId);
+      let value;
+
+      switch (settingKey) {
+        case 'Json_Data':
+          value = JSON.stringify(response.setting || '', null, '   ');
+          break;
+        default:
+          value = response.setting || ''
+      }
+
+      this.setState({
+        settingResult: response,
+        settingValue: value
+      });
+    });
+  }
+
+  // Handler to get settings value into state, when you modify textarea element in the dialog
+  handleSettingsValueChange(event: any) {
+    const settingValue = event.target.value;
+    this.setState({ settingValue: settingValue });
+  }
+
+  // The showcase does not have permission to write data, it is expected to fail with 403 Forbidden. 
+  // However saveSetting method will work in your project with signed-in user, who has required permissions in the project.
+  saveSettings() {
+    this.setState({ saveInProgress: true });
+    AuthorizedFrontendRequestContext.create().then(async context => {
+      const response = await IModelApp.settings.saveSetting(context, this.state.settingValue, namespace, this.state.settingKey!, true, this.state.projectId, this.state.imodelId);
+      this.setState({
+        settingResult: response,
+        saveInProgress: false
+      });
+    });
+  }
+
+  // Helper method to show status get/write operations with external setting in the dialog
+  private showStatus() {
+    if (!this.state.settingResult || this.state.settingResult.status == SettingsStatus.Success) {
+      return (<div></div>);
+    }
+
+    return (
+      <div style={{ lineBreak: 'anywhere', overflowWrap: 'break-word', maxWidth: '30vw' }}>
+        <SmallText style={{ color: 'var(--foreground-alert)' }}>
+          {`${this.state.settingResult.status} ${this.state.settingResult.errorMessage}`}
+        </SmallText>
+      </div>
+    );
+  }
+
+  /** Components for rendering the sample's instructions and controls */
+  private getControls() {
+    return (
+      <>
+        <LabeledSelect label="Settings Name:" options={["Json_Data", "Arbitrary_Text", "CSV_Data"]} placeholder="Browse by name..." value={this.state.settingKey} onChange={this.handleSettingsChange} />
```

Comment:
The styling of controls in not consistent with the other samples.  For example your LabeledSelect has a different background color from the Select controls used elsewhere.

Reply:
LabeledSelect has default style as displayed in UI Components section. Other samples use plain unstyled "select", not sure why. I can unify that in another pull request

Reply:
Made dialog to be visually consistent with other examples